### ISSUE TITLE
ConnectionRefusedError: [Errno 111] Connection refused

### ISSUE THREAD

**Issue by kwispykweems2** [YoRyan/nuxhash#22]
I'm trying to get this to function on a minimal headless Centos 7 server, but it seems to fail in various ways.  ConnectionRefusedError on initial start up with no settings.conf.  If I run nuxhashd again, putting in my info, it just hangs, CPU at 100%.  I've tried various arguments, -v and --show-mining, but I can't figure out why the program fails. (I've run Claymore ETH miner on this before without an issue, and now I want to try other algorithms)

$ nuxhashd
nuxhashd initial setup
Wallet address: 3QtrjJpp9HYcgkJf7omZFMJZ6s3ELuxbot
Worker name: drew
Region (eu/usa/hk/jp/in/br): usa

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   617    0   617    0     0   1577      0 --:--:-- --:--:-- --:--:--  1582
100 5830k  100 5830k    0     0  3190k      0  0:00:01  0:00:01 --:--:-- 4360k
Traceback (most recent call last):
  File "/opt/rh/rh-python36/root/usr/bin/nuxhashd", line 11, in <module>
    load_entry_point('nuxhash==1.0.0b2', 'console_scripts', 'nuxhashd')()
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/nuxhash/__init__.py", line 3, in nuxhashd
    main()
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/nuxhash/daemon.py", line 105, in main
    nx_miners, nx_settings, all_devices, nx_benchmarks)
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/nuxhash/daemon.py", line 145, in run_missing_benchmarks
    miner.load()
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/nuxhash/miners/excavator.py", line 380, in load
    self.server.start()
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/nuxhash/miners/excavator.py", line 127, in start
    self._read_devices()
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/nuxhash/miners/excavator.py", line 209, in _read_devices
    response = self.send_command('device.list', [])
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/nuxhash/miners/excavator.py", line 167, in send_command
    with socket.create_connection(self._address, self.TIMEOUT) as s:
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/socket.py", line 724, in create_connection
    raise err
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/socket.py", line 713, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused
2019-03-31 22:44:54,203 CRITICAL: Crash! Killing all miners.
Killed
[drew@drewshbag ~]$ nuxhashd --show-mining
nuxhashd initial setup
Wallet address: 3QtrjJpp9HYcgkJf7omZFMJZ6s3ELuxbot
Worker name: drew
Region (eu/usa/hk/jp/in/br): usa

Since Centos 7 only has python 2.7, I had to install python 3.6.3 with a Software Collection (https://linuxize.com/post/how-to-install-python-3-on-centos-7/), hopefully that isn't causing an issue.
`$ python -V
Python 3.6.3`

After looking through the Issues, I added a settings.conf with a [nicehash] section.  I ran nuxhashd a few more times, failing or hanging.  Then I checked my settings.conf and nuxhashd had written a few more sections: (I added the listen address and port)
My .config/nuxhash/settings.conf:
`[nicehash]
wallet = 3QtrjJpp9HYcgkJf7omZFMJZ6s3ELuxbot
workername = nuxhash
region = usa

[switching]
interval = 60
threshold = 0.1

[gui]
units = mBTC

[donate]
optout = False

[excavator_miner]
listen = 127.0.0.1:4000
args =`

I don't know if nicehash broke something on their end, but I can't get nicehash to mine on my setup.  Any help?

**Comment by YoRyan** [comment]
Apologies for the delay. My hunch is that excavator is failing to start. Run `~/.config/nuxhash/excavator/excavator -i 127.0.0.1 -p 5001` in a terminal and see what happens.

**Comment by kwispykweems2** [comment]
Problem was on my end.  I was using outdated Nvidia drivers (390) somehow on my CentOS install.  When I ran just excavator like you said, it just said "couldn't find any devices" and quit.  Some googling led my to believe it was an openCL issue or outdated issue.
After I did a full "yum upgrade" of my server, having seen old drivers and kernel present, excavator now works great!

$ nuxhashd --show-mining

CUDA device: GeForce GTX 1060 3GB (GPU-608d17b7-876b-d3a5-8b0b-cfe98eaea989)
  excavator_x16r:   6.26 MH/s                      
  excavator_cryptonightV7: 415.77  H/s                      
  excavator_daggerhashimoto_decred:  15.39 MH/s, 369.24 MH/s                      
  excavator_keccak .   478.75 MH/s (sampling, 54 s)
